docs/troubleshooting.md

```diff
@@ -62,3 +63,79 @@ For instance, try this:
 ```
 
+## Adding a new Certificate Authority to a cluster that has become unreachable
+
+If you have recently added or replaced the API certificate of your cluster, and it has become unreachable, this is the SOP for you.  Usually this will manifest as an error message similar to: `x509: certificate signed by unknown authority`.  There are several ways to regain access to a cluster in this state.
+
+In all cases, you will need the certificate authority (CA) for your new API certificate in standard PEM format.  It should look something like this:
+
+```sh
+$ cat new-ca-certificate.pem
+```
+
+```
+-----BEGIN CERTIFICATE-----
+<base64 encoded certificate blob>
+-----END CERTIFICATE-----
+...
+-----BEGIN CERTIFICATE-----
+<potentially more base64 encoded certificate blobs>
+-----END CERTIFICATE-----
+```
+
+### Replacing the API certificate of a managed single cluster with kubeconfig
+
+In order for a new API certificate to be accepted by the system, the CA certificate used to sign the new API certificate must be added to the kubeconfig, alongside or replacing any existing certificates. Start by inspecting your current kubeconfig:
+
+```sh
+$ export MY_KUBECONFIG_NAME=$(oc get cd $CLUSTERNAME -o "jsonpath={.spec.clusterMetadata.adminKubeconfigSecretRef.name}")
+$ oc extract secret/$MY_KUBECONFIG_NAME --keys=kubeconfig --to=- > my-kubeconfig-file
+$ cat my-kubeconfig-file
+```
+
+```yaml
+...
+clusters:
+- cluster:
+    certificate-authority-data: <base64 encoded blob>
+    server: https://api.<your cluster>.<your domain>:6443
+  name: <your cluster>
+...
+```
+
+The field `certificate-authority-data` is what needs to be updated. When base64 decoded, it looks something like this:
+
+```sh
+$ echo <base64 encoded blob> | base64 --decode > decoded-existing-certs.pem
+$ cat decoded-existing-certs.pem
+```
+
+```
+-----BEGIN CERTIFICATE-----
+<base64 encoded certificate blob>
+-----END CERTIFICATE-----
+...
+-----BEGIN CERTIFICATE-----
+<potentially more base64 encoded certificate blobs>
+-----END CERTIFICATE-----
+```
+
+Concatenate your new CA certificate with the existing decoded CA certificates:
+
+```sh
+$ cat decoded-existing-certs.pem new-ca-certificate.pem | openssl base64 -A
+```
+
+Copy the contents of this new base64 encoded blob and use it to replace the `certificate-authority-data` field in your kubeconfig. After the kubeconfig file is ready, patch your cluster to use the new kubeconfig:
+
+```sh
+oc patch secret $MY_KUBECONFIG_NAME --type='json' -p="[{'op': 'replace', 'path': '/data/kubeconfig', 'value': '$(openssl base64 -A -in my-updated-kubeconfig-file)'}]"
+```
+
+### Including a new CA certificate globally on the managing cluster with hiveconfig
+
+Hive can use its own configuration file to configure an additional CA certificate which will be added to kubeconfigs referenced by _ALL_ ClusterDeployments. Use the `hack/set-additional-ca.sh` script to accomplish this:
+
+```sh
```

Comment:
Please note that the command `oc patch` needs to be updated based on this interaction in the original issue:
https://github.com/openshift/hive/issues/1900#issuecomment-1333929786

This would need confirmation from your test automation, but I believe the command should be:

```sh
oc patch secret $MY_KUBECONFIG_NAME --type='json' -p="[{'op': 'replace', 'path': '/data/kubeconfig', 'value': '$(openssl base64 -A -in my-updated-kubeconfig-file)'},{'op': 'replace', 'path': '/data/raw-kubeconfig', 'value': '$(openssl base64 -A -in my-updated-kubeconfig-file)'}]"
```